kong/clustering/wrpc_data_plane.lua

```diff
@@ -43,90 +50,90 @@ function _M.new(conf, cert, cert_key)
   return setmetatable(self, _MT)
 end
 
+local communicate
 
 function _M:init_worker(plugins_list)
   -- ROLE = "data_plane"
 
   self.plugins_list = plugins_list
 
   if ngx.worker.id() == 0 then
-    assert(ngx.timer.at(0, function(premature)
-      self:communicate(premature)
-    end))
+    communicate(self)
   end
 end
 
 
-local wrpc_config_service
-local function get_config_service()
-  if not wrpc_config_service then
-    wrpc_config_service = wrpc_proto.new()
-    wrpc_config_service:import("kong.services.config.v1.config")
-    wrpc_config_service:set_handler("ConfigService.SyncConfig", function(peer, data)
-      -- yield between steps to prevent long delay
-      if peer.config_semaphore then
-        local json_config = assert(inflate_gzip(data.config))
-        yield()
-        peer.config_obj.next_config = assert(cjson_decode(json_config))
-        yield()
-
-        peer.config_obj.next_hash = data.config_hash
-        peer.config_obj.next_hashes = data.hashes
-        peer.config_obj.next_config_version = tonumber(data.version)
-        if peer.config_semaphore:count() <= 0 then
-          -- the following line always executes immediately after the `if` check
-          -- because `:count` will never yield, end result is that the semaphore
-          -- count is guaranteed to not exceed 1
-          peer.config_semaphore:post()
-        end
+local function init_config_service(service)
+  service:import("kong.services.config.v1.config")
+  service:set_handler("ConfigService.SyncConfig", function(peer, data)
+    -- yield between steps to prevent long delay
+    if peer.config_semaphore then
+      local json_config = assert(inflate_gzip(data.config))
+      yield()
+      peer.config_obj.next_config = assert(cjson_decode(json_config))
+      yield()
+
+      peer.config_obj.next_hash = data.config_hash
+      peer.config_obj.next_hashes = data.hashes
+      peer.config_obj.next_config_version = tonumber(data.version)
+      if peer.config_semaphore:count() <= 0 then
+        -- the following line always executes immediately after the `if` check
+        -- because `:count` will never yield, end result is that the semaphore
+        -- count is guaranteed to not exceed 1
+        peer.config_semaphore:post()
       end
-      return { accepted = true }
-    end)
-  end
-
-  return wrpc_config_service
+    end
+    return { accepted = true }
+  end)
 end
 
-function _M:communicate(premature)
-  if premature then
-    -- worker wants to exit
-    return
+local wrpc_services
+local function get_services()
+  if not wrpc_services then
+    wrpc_services = wrpc_proto.new()
+    init_negotiation_client(wrpc_services)
+    init_config_service(wrpc_services)
   end
 
-  local conf = self.conf
+  return wrpc_services
+end
+
+local function communicate_body(cp)
```

Comment:
`communicate_body` is a little strange, could we change it to `communicate_impl` or others?